### PR TITLE
First working version, ready for v1.0

### PR DESCRIPTION
**Pull request by michaelaw320**
Tested most examples are working as intended

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Run pre-commit test
+      run: |
+        curl https://pre-commit.com/install-local.py | python3 -
+        curl -L "$(curl -s https://api.github.com/repos/segmentio/terraform-docs/releases/latest | grep -o -E "https://.+?-linux-amd64")" > terraform-docs && chmod +x terraform-docs && sudo mv terraform-docs /usr/bin/
```
Comment (dimaqq) [inline .github/workflows/lint.yml]: Maybe use `jq` to select the value from json?
My gut reaction is that there may be several lines that match `-linux-amd64` and then it's not clear what the script will download...

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: lint
```
Comment (dimaqq) [inline .github/workflows/lint.yml]: Is the workflow automatically started, or does it need to be referenced somewhere?

File: .github/workflows/lint.yml (hunk 1 above)
Comment (michaelaw320) [reply .github/workflows/lint.yml]: It's written like that in the official documentation to install

File: .github/workflows/lint.yml (hunk 2 above)
Comment (michaelaw320) [reply .github/workflows/lint.yml]: On Line 3, it will start on any push event

**Pull request merged**